tests/functional/test_install_wheel.py

```diff
@@ -534,3 +534,34 @@ def test_wheel_installs_ok_with_nested_dist_info(script):
     script.pip(
         "install", "--no-cache-dir", "--no-index", package
     )
+
+
+def test_wheel_installs_ok_with_badly_encoded_irrelevant_metadata(script):
```

Comment:
```suggestion
def test_wheel_installs_ok_with_badly_encoded_irrelevant_file(script):
```

Reply:
"metadata" here was implying "in the `.dist-info` directory" (see also `Distribution.get_metadata()`), whereas "file" to me could be any old file. Probably "dist_info_file" would be a good compromise.

Reply:
Sounds good to me. :)

The reason for the suggestion was "we're not loading the METADATA file".

Reply:
The `IMetadataProvider` interface is indeed confusing in that respect.

Reply:
Updated [here](https://github.com/pypa/pip/pull/7539/files#diff-f13a380a0a5a8c7904168723506d1fb4R539) to `test_wheel_installs_ok_with_badly_encoded_irrelevant_dist_info_file`